### PR TITLE
Thomas/migration2

### PR DESCRIPTION
**Pull request by thomasgonda3**

### CODE REVIEW

File: src/bin/migration.ts [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+  asNumber,
+  asObject,
+  asOptional,
+  asString,
+  asUnknown
+} from 'cleaners'
+import js from 'jsonfile'
+import nano from 'nano'
+
+import config from '../../config.json'
```
Comment (thomasgonda3) [inline src/bin/migration.ts:15]: use fs to import config file

File: src/bin/migration.ts [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    const appAndPluginId = `edge_${partner.name}`
+    const query = {
+      selector: {
+        timestamp: { $gte: 0 }
+      },
+      fields: ['timestamp'],
+      limit: 100000000
+    }
+    const transactionTimestamps = asTimestampResult(
+      await reportsTransactions.partitionedFind(appAndPluginId, query)
```
Comment (thomasgonda3) [inline src/bin/migration.ts:131]: partitionedList

File: src/bin/migration.ts [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    )
+
+    let earliestTimestamp = transactionTimestamps.docs[0].timestamp
+    for (let i = 1; i < transactionTimestamps.docs.length; i++) {
+      earliestTimestamp =
```
Comment (thomasgonda3) [inline src/bin/migration.ts:136]: do if instead of tertiary

File: src/bin/migration.ts [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+        timestamp,
+        usdValue: undefined,
+        rawTx: undefined
+      }
+    })
+    datelog(
+      `Importing ${filteredTransactions.length} transactions for ${partner.name} before date ${earliestDate}.`
+    )
+    const transactionsArray: StandardTx[] = []
+    for (const transaction of reformattedTxs) {
```
Comment (thomasgonda3) [inline src/bin/migration.ts:216]: include everything from line 216-236 into .map

**Review by paullinator — APPROVED** [review]

**Pull request merged**